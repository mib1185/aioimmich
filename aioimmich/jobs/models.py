"""aioimmich jobs models."""

from __future__ import annotations

from dataclasses import dataclass, field

from mashumaro import field_options
from mashumaro.mixins.json import DataClassJSONMixin


@dataclass
class JobCounts(DataClassJSONMixin):
    """Representation of job counts for a specific job type."""

    active: int
    completed: int
    delayed: int
    failed: int
    paused: int
    waiting: int


@dataclass
class QueueStatus(DataClassJSONMixin):
    """Representation of queue status for a specific job type."""

    is_active: bool = field(metadata=field_options(alias="isActive"))
    is_paused: bool = field(metadata=field_options(alias="isPaused"))


@dataclass
class JobStatus(DataClassJSONMixin):
    """Representation of status for a specific job type."""

    job_counts: JobCounts = field(metadata=field_options(alias="jobCounts"))
    queue_status: QueueStatus = field(metadata=field_options(alias="queueStatus"))


@dataclass
class ImmichJobsStatus(DataClassJSONMixin):
    """Representation of all jobs status."""

    background_task: JobStatus = field(metadata=field_options(alias="backgroundTask"))
    backup_database: JobStatus = field(metadata=field_options(alias="backupDatabase"))
    duplicate_detection: JobStatus = field(
        metadata=field_options(alias="duplicateDetection")
    )
    face_detection: JobStatus = field(metadata=field_options(alias="faceDetection"))
    facial_recognition: JobStatus = field(
        metadata=field_options(alias="facialRecognition")
    )
    library: JobStatus
    metadata_extraction: JobStatus = field(
        metadata=field_options(alias="metadataExtraction")
    )
    migration: JobStatus
    notifications: JobStatus
    search: JobStatus
    sidecar: JobStatus
    smart_search: JobStatus = field(metadata=field_options(alias="smartSearch"))
    storage_template_migration: JobStatus = field(
        metadata=field_options(alias="storageTemplateMigration")
    )
    thumbnail_generation: JobStatus = field(
        metadata=field_options(alias="thumbnailGeneration")
    )
    video_conversion: JobStatus = field(metadata=field_options(alias="videoConversion"))


@dataclass
class JobCommand(DataClassJSONMixin):
    """Representation of a job command request."""

    command: str
    force: bool = False
