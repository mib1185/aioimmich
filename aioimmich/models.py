"""aioimmich models."""

from __future__ import annotations

from dataclasses import dataclass


@dataclass
class ImmichAsset:
    """Representation of an immich asset."""

    asset_id: str
    file_name: str
    mime_type: str


@dataclass
class ImmichAlbum:
    """Representation of an immich album."""

    album_id: str
    name: str
    description: str
    thumbnail_asset_id: str
    asset_count: int
    assets: list[ImmichAsset]
