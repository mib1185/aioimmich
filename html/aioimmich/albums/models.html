<!doctype html>
<html lang="en">
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1, minimum-scale=1">
<meta name="generator" content="pdoc3 0.11.6">
<title>aioimmich.albums.models API documentation</title>
<meta name="description" content="aioimmich albums models.">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/13.0.0/sanitize.min.css" integrity="sha512-y1dtMcuvtTMJc1yPgEqF0ZjQbhnc/bFhyvIyVNb9Zk5mIGtqVaAB1Ttl28su8AvFMOY0EwRbAe+HCLqj6W7/KA==" crossorigin>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/10up-sanitize.css/13.0.0/typography.min.css" integrity="sha512-Y1DYSb995BAfxobCkKepB1BqJJTPrOp3zPL74AWFugHHmmdcvO+C48WLrUOlhGMc0QG7AE3f7gmvvcrmX2fDoA==" crossorigin>
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.9.0/styles/default.min.css" crossorigin>
<style>:root{--highlight-color:#fe9}.flex{display:flex !important}body{line-height:1.5em}#content{padding:20px}#sidebar{padding:1.5em;overflow:hidden}#sidebar > *:last-child{margin-bottom:2cm}.http-server-breadcrumbs{font-size:130%;margin:0 0 15px 0}#footer{font-size:.75em;padding:5px 30px;border-top:1px solid #ddd;text-align:right}#footer p{margin:0 0 0 1em;display:inline-block}#footer p:last-child{margin-right:30px}h1,h2,h3,h4,h5{font-weight:300}h1{font-size:2.5em;line-height:1.1em}h2{font-size:1.75em;margin:2em 0 .50em 0}h3{font-size:1.4em;margin:1.6em 0 .7em 0}h4{margin:0;font-size:105%}h1:target,h2:target,h3:target,h4:target,h5:target,h6:target{background:var(--highlight-color);padding:.2em 0}a{color:#058;text-decoration:none;transition:color .2s ease-in-out}a:visited{color:#503}a:hover{color:#b62}.title code{font-weight:bold}h2[id^="header-"]{margin-top:2em}.ident{color:#900;font-weight:bold}pre code{font-size:.8em;line-height:1.4em;padding:1em;display:block}code{background:#f3f3f3;font-family:"DejaVu Sans Mono",monospace;padding:1px 4px;overflow-wrap:break-word}h1 code{background:transparent}pre{border-top:1px solid #ccc;border-bottom:1px solid #ccc;margin:1em 0}#http-server-module-list{display:flex;flex-flow:column}#http-server-module-list div{display:flex}#http-server-module-list dt{min-width:10%}#http-server-module-list p{margin-top:0}.toc ul,#index{list-style-type:none;margin:0;padding:0}#index code{background:transparent}#index h3{border-bottom:1px solid #ddd}#index ul{padding:0}#index h4{margin-top:.6em;font-weight:bold}@media (min-width:200ex){#index .two-column{column-count:2}}@media (min-width:300ex){#index .two-column{column-count:3}}dl{margin-bottom:2em}dl dl:last-child{margin-bottom:4em}dd{margin:0 0 1em 3em}#header-classes + dl > dd{margin-bottom:3em}dd dd{margin-left:2em}dd p{margin:10px 0}.name{background:#eee;font-size:.85em;padding:5px 10px;display:inline-block;min-width:40%}.name:hover{background:#e0e0e0}dt:target .name{background:var(--highlight-color)}.name > span:first-child{white-space:nowrap}.name.class > span:nth-child(2){margin-left:.4em}.inherited{color:#999;border-left:5px solid #eee;padding-left:1em}.inheritance em{font-style:normal;font-weight:bold}.desc h2{font-weight:400;font-size:1.25em}.desc h3{font-size:1em}.desc dt code{background:inherit}.source > summary,.git-link-div{color:#666;text-align:right;font-weight:400;font-size:.8em;text-transform:uppercase}.source summary > *{white-space:nowrap;cursor:pointer}.git-link{color:inherit;margin-left:1em}.source pre{max-height:500px;overflow:auto;margin:0}.source pre code{font-size:12px;overflow:visible;min-width:max-content}.hlist{list-style:none}.hlist li{display:inline}.hlist li:after{content:',\2002'}.hlist li:last-child:after{content:none}.hlist .hlist{display:inline;padding-left:1em}img{max-width:100%}td{padding:0 .5em}.admonition{padding:.1em 1em;margin:1em 0}.admonition-title{font-weight:bold}.admonition.note,.admonition.info,.admonition.important{background:#aef}.admonition.todo,.admonition.versionadded,.admonition.tip,.admonition.hint{background:#dfd}.admonition.warning,.admonition.versionchanged,.admonition.deprecated{background:#fd4}.admonition.error,.admonition.danger,.admonition.caution{background:lightpink}</style>
<style media="screen and (min-width: 700px)">@media screen and (min-width:700px){#sidebar{width:30%;height:100vh;overflow:auto;position:sticky;top:0}#content{width:70%;max-width:100ch;padding:3em 4em;border-left:1px solid #ddd}pre code{font-size:1em}.name{font-size:1em}main{display:flex;flex-direction:row-reverse;justify-content:flex-end}.toc ul ul,#index ul ul{padding-left:1em}.toc > ul > li{margin-top:.5em}}</style>
<style media="print">@media print{#sidebar h1{page-break-before:always}.source{display:none}}@media print{*{background:transparent !important;color:#000 !important;box-shadow:none !important;text-shadow:none !important}a[href]:after{content:" (" attr(href) ")";font-size:90%}a[href][title]:after{content:none}abbr[title]:after{content:" (" attr(title) ")"}.ir a:after,a[href^="javascript:"]:after,a[href^="#"]:after{content:""}pre,blockquote{border:1px solid #999;page-break-inside:avoid}thead{display:table-header-group}tr,img{page-break-inside:avoid}img{max-width:100% !important}@page{margin:0.5cm}p,h2,h3{orphans:3;widows:3}h1,h2,h3,h4,h5,h6{page-break-after:avoid}}</style>
<script defer src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/11.9.0/highlight.min.js" integrity="sha512-D9gUyxqja7hBtkWpPWGt9wfbfaMGVt9gnyCvYa+jojwwPHLCzUm5i8rpk7vD7wNee9bA35eYIjobYPaQuKS1MQ==" crossorigin></script>
<script>window.addEventListener('DOMContentLoaded', () => {
hljs.configure({languages: ['bash', 'css', 'diff', 'graphql', 'ini', 'javascript', 'json', 'plaintext', 'python', 'python-repl', 'rust', 'shell', 'sql', 'typescript', 'xml', 'yaml']});
hljs.highlightAll();
/* Collapse source docstrings */
setTimeout(() => {
[...document.querySelectorAll('.hljs.language-python > .hljs-string')]
.filter(el => el.innerHTML.length > 200 && ['"""', "'''"].includes(el.innerHTML.substring(0, 3)))
.forEach(el => {
let d = document.createElement('details');
d.classList.add('hljs-string');
d.innerHTML = '<summary>"""</summary>' + el.innerHTML.substring(3);
el.replaceWith(d);
});
}, 100);
})</script>
</head>
<body>
<main>
<article id="content">
<header>
<h1 class="title">Module <code>aioimmich.albums.models</code></h1>
</header>
<section id="section-intro">
<p>aioimmich albums models.</p>
</section>
<section>
</section>
<section>
</section>
<section>
</section>
<section>
<h2 class="section-title" id="header-classes">Classes</h2>
<dl>
<dt id="aioimmich.albums.models.AssetToAlbumError"><code class="flex name class">
<span>class <span class="ident">AssetToAlbumError</span></span>
<span>(</span><span>*args, **kwds)</span>
</code></dt>
<dd>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">class AssetToAlbumError(StrEnum):
    &#34;&#34;&#34;Asset upload status.&#34;&#34;&#34;

    DUPLICATED = &#34;duplicate&#34;
    NO_PERMISSION = &#34;no_permission&#34;
    NOT_FOUND = &#34;not_found&#34;
    UNKNOWN = &#34;unknown&#34;</code></pre>
</details>
<div class="desc"><p>Asset upload status.</p></div>
<h3>Ancestors</h3>
<ul class="hlist">
<li>enum.StrEnum</li>
<li>builtins.str</li>
<li>enum.ReprEnum</li>
<li>enum.Enum</li>
</ul>
<h3>Class variables</h3>
<dl>
<dt id="aioimmich.albums.models.AssetToAlbumError.DUPLICATED"><code class="name">var <span class="ident">DUPLICATED</span></code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.AssetToAlbumError.NOT_FOUND"><code class="name">var <span class="ident">NOT_FOUND</span></code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.AssetToAlbumError.NO_PERMISSION"><code class="name">var <span class="ident">NO_PERMISSION</span></code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.AssetToAlbumError.UNKNOWN"><code class="name">var <span class="ident">UNKNOWN</span></code></dt>
<dd>
<div class="desc"></div>
</dd>
</dl>
</dd>
<dt id="aioimmich.albums.models.ImmichAddAssetsToAlbumResponse"><code class="flex name class">
<span>class <span class="ident">ImmichAddAssetsToAlbumResponse</span></span>
<span>(</span><span>asset_id: str,<br>success: bool,<br>error: <a title="aioimmich.albums.models.AssetToAlbumError" href="#aioimmich.albums.models.AssetToAlbumError">AssetToAlbumError</a> | None = None)</span>
</code></dt>
<dd>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">@dataclass
class ImmichAddAssetsToAlbumResponse(DataClassJSONMixin):
    &#34;&#34;&#34;Representation of an immich add assets to album response.&#34;&#34;&#34;

    asset_id: str = field(metadata=field_options(alias=&#34;id&#34;))
    success: bool
    error: AssetToAlbumError | None = field(default=None)</code></pre>
</details>
<div class="desc"><p>Representation of an immich add assets to album response.</p></div>
<h3>Ancestors</h3>
<ul class="hlist">
<li>mashumaro.mixins.json.DataClassJSONMixin</li>
<li>mashumaro.mixins.dict.DataClassDictMixin</li>
</ul>
<h3>Static methods</h3>
<dl>
<dt id="aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.from_dict"><code class="name flex">
<span>def <span class="ident">from_dict</span></span>(<span>d, *, dialect=None)</span>
</code></dt>
<dd>
<div class="desc"></div>
</dd>
</dl>
<h3>Instance variables</h3>
<dl>
<dt id="aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.asset_id"><code class="name">var <span class="ident">asset_id</span> : str</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.error"><code class="name">var <span class="ident">error</span> : <a title="aioimmich.albums.models.AssetToAlbumError" href="#aioimmich.albums.models.AssetToAlbumError">AssetToAlbumError</a> | None</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.success"><code class="name">var <span class="ident">success</span> : bool</code></dt>
<dd>
<div class="desc"></div>
</dd>
</dl>
<h3>Methods</h3>
<dl>
<dt id="aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.to_dict"><code class="name flex">
<span>def <span class="ident">to_dict</span></span>(<span>self)</span>
</code></dt>
<dd>
<div class="desc"></div>
</dd>
</dl>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum"><code class="flex name class">
<span>class <span class="ident">ImmichAlbum</span></span>
<span>(</span><span>album_id: str,<br>album_name: str,<br>album_thumbnail_asset_id: str | None,<br>album_users: list,<br>asset_count: int,<br>assets: list[ImmichAsset],<br>created_at: datetime,<br>has_shared_link: bool,<br>is_activity_enabled: bool,<br>owner_id: str,<br>owner: ImmichUser,<br>shared: bool,<br>updated_at: datetime,<br>end_date: datetime | None = None,<br>last_modified_asset_timestamp: datetime | None = None,<br>order: str | None = None,<br>start_date: datetime | None = None)</span>
</code></dt>
<dd>
<details class="source">
<summary>
<span>Expand source code</span>
</summary>
<pre><code class="python">@dataclass
class ImmichAlbum(DataClassJSONMixin):
    &#34;&#34;&#34;Representation of an immich album.&#34;&#34;&#34;

    # non-default parameters
    album_id: str = field(metadata=field_options(alias=&#34;id&#34;))
    album_name: str = field(metadata=field_options(alias=&#34;albumName&#34;))
    album_thumbnail_asset_id: str | None = field(
        metadata=field_options(alias=&#34;albumThumbnailAssetId&#34;)
    )
    album_users: list = field(metadata=field_options(alias=&#34;albumUsers&#34;))
    asset_count: int = field(metadata=field_options(alias=&#34;assetCount&#34;))
    assets: list[ImmichAsset]
    created_at: datetime = field(metadata=field_options(alias=&#34;createdAt&#34;))
    has_shared_link: bool = field(metadata=field_options(alias=&#34;hasSharedLink&#34;))
    is_activity_enabled: bool = field(metadata=field_options(alias=&#34;isActivityEnabled&#34;))
    owner_id: str = field(metadata=field_options(alias=&#34;ownerId&#34;))
    owner: ImmichUser
    shared: bool
    updated_at: datetime = field(metadata=field_options(alias=&#34;updatedAt&#34;))

    # default parameters
    end_date: datetime | None = field(
        metadata=field_options(alias=&#34;endDate&#34;), default=None
    )
    last_modified_asset_timestamp: datetime | None = field(
        metadata=field_options(alias=&#34;lastModifiedAssetTimestamp&#34;), default=None
    )
    order: str | None = field(default=None)
    start_date: datetime | None = field(
        metadata=field_options(alias=&#34;startDate&#34;), default=None
    )</code></pre>
</details>
<div class="desc"><p>Representation of an immich album.</p></div>
<h3>Ancestors</h3>
<ul class="hlist">
<li>mashumaro.mixins.json.DataClassJSONMixin</li>
<li>mashumaro.mixins.dict.DataClassDictMixin</li>
</ul>
<h3>Static methods</h3>
<dl>
<dt id="aioimmich.albums.models.ImmichAlbum.from_dict"><code class="name flex">
<span>def <span class="ident">from_dict</span></span>(<span>d, *, dialect=None)</span>
</code></dt>
<dd>
<div class="desc"></div>
</dd>
</dl>
<h3>Instance variables</h3>
<dl>
<dt id="aioimmich.albums.models.ImmichAlbum.album_id"><code class="name">var <span class="ident">album_id</span> : str</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.album_name"><code class="name">var <span class="ident">album_name</span> : str</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.album_thumbnail_asset_id"><code class="name">var <span class="ident">album_thumbnail_asset_id</span> : str | None</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.album_users"><code class="name">var <span class="ident">album_users</span> : list</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.asset_count"><code class="name">var <span class="ident">asset_count</span> : int</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.assets"><code class="name">var <span class="ident">assets</span> : list[<a title="aioimmich.assets.models.ImmichAsset" href="../assets/models.html#aioimmich.assets.models.ImmichAsset">ImmichAsset</a>]</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.created_at"><code class="name">var <span class="ident">created_at</span> : datetime.datetime</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.end_date"><code class="name">var <span class="ident">end_date</span> : datetime.datetime | None</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.has_shared_link"><code class="name">var <span class="ident">has_shared_link</span> : bool</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.is_activity_enabled"><code class="name">var <span class="ident">is_activity_enabled</span> : bool</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.last_modified_asset_timestamp"><code class="name">var <span class="ident">last_modified_asset_timestamp</span> : datetime.datetime | None</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.order"><code class="name">var <span class="ident">order</span> : str | None</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.owner"><code class="name">var <span class="ident">owner</span> : <a title="aioimmich.users.models.ImmichUser" href="../users/models.html#aioimmich.users.models.ImmichUser">ImmichUser</a></code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.owner_id"><code class="name">var <span class="ident">owner_id</span> : str</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.shared"><code class="name">var <span class="ident">shared</span> : bool</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.start_date"><code class="name">var <span class="ident">start_date</span> : datetime.datetime | None</code></dt>
<dd>
<div class="desc"></div>
</dd>
<dt id="aioimmich.albums.models.ImmichAlbum.updated_at"><code class="name">var <span class="ident">updated_at</span> : datetime.datetime</code></dt>
<dd>
<div class="desc"></div>
</dd>
</dl>
<h3>Methods</h3>
<dl>
<dt id="aioimmich.albums.models.ImmichAlbum.to_dict"><code class="name flex">
<span>def <span class="ident">to_dict</span></span>(<span>self)</span>
</code></dt>
<dd>
<div class="desc"></div>
</dd>
</dl>
</dd>
</dl>
</section>
</article>
<nav id="sidebar">
<div class="toc">
<ul></ul>
</div>
<ul id="index">
<li><h3>Super-module</h3>
<ul>
<li><code><a title="aioimmich.albums" href="index.html">aioimmich.albums</a></code></li>
</ul>
</li>
<li><h3><a href="#header-classes">Classes</a></h3>
<ul>
<li>
<h4><code><a title="aioimmich.albums.models.AssetToAlbumError" href="#aioimmich.albums.models.AssetToAlbumError">AssetToAlbumError</a></code></h4>
<ul class="">
<li><code><a title="aioimmich.albums.models.AssetToAlbumError.DUPLICATED" href="#aioimmich.albums.models.AssetToAlbumError.DUPLICATED">DUPLICATED</a></code></li>
<li><code><a title="aioimmich.albums.models.AssetToAlbumError.NOT_FOUND" href="#aioimmich.albums.models.AssetToAlbumError.NOT_FOUND">NOT_FOUND</a></code></li>
<li><code><a title="aioimmich.albums.models.AssetToAlbumError.NO_PERMISSION" href="#aioimmich.albums.models.AssetToAlbumError.NO_PERMISSION">NO_PERMISSION</a></code></li>
<li><code><a title="aioimmich.albums.models.AssetToAlbumError.UNKNOWN" href="#aioimmich.albums.models.AssetToAlbumError.UNKNOWN">UNKNOWN</a></code></li>
</ul>
</li>
<li>
<h4><code><a title="aioimmich.albums.models.ImmichAddAssetsToAlbumResponse" href="#aioimmich.albums.models.ImmichAddAssetsToAlbumResponse">ImmichAddAssetsToAlbumResponse</a></code></h4>
<ul class="">
<li><code><a title="aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.asset_id" href="#aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.asset_id">asset_id</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.error" href="#aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.error">error</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.from_dict" href="#aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.from_dict">from_dict</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.success" href="#aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.success">success</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.to_dict" href="#aioimmich.albums.models.ImmichAddAssetsToAlbumResponse.to_dict">to_dict</a></code></li>
</ul>
</li>
<li>
<h4><code><a title="aioimmich.albums.models.ImmichAlbum" href="#aioimmich.albums.models.ImmichAlbum">ImmichAlbum</a></code></h4>
<ul class="">
<li><code><a title="aioimmich.albums.models.ImmichAlbum.album_id" href="#aioimmich.albums.models.ImmichAlbum.album_id">album_id</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.album_name" href="#aioimmich.albums.models.ImmichAlbum.album_name">album_name</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.album_thumbnail_asset_id" href="#aioimmich.albums.models.ImmichAlbum.album_thumbnail_asset_id">album_thumbnail_asset_id</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.album_users" href="#aioimmich.albums.models.ImmichAlbum.album_users">album_users</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.asset_count" href="#aioimmich.albums.models.ImmichAlbum.asset_count">asset_count</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.assets" href="#aioimmich.albums.models.ImmichAlbum.assets">assets</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.created_at" href="#aioimmich.albums.models.ImmichAlbum.created_at">created_at</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.end_date" href="#aioimmich.albums.models.ImmichAlbum.end_date">end_date</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.from_dict" href="#aioimmich.albums.models.ImmichAlbum.from_dict">from_dict</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.has_shared_link" href="#aioimmich.albums.models.ImmichAlbum.has_shared_link">has_shared_link</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.is_activity_enabled" href="#aioimmich.albums.models.ImmichAlbum.is_activity_enabled">is_activity_enabled</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.last_modified_asset_timestamp" href="#aioimmich.albums.models.ImmichAlbum.last_modified_asset_timestamp">last_modified_asset_timestamp</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.order" href="#aioimmich.albums.models.ImmichAlbum.order">order</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.owner" href="#aioimmich.albums.models.ImmichAlbum.owner">owner</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.owner_id" href="#aioimmich.albums.models.ImmichAlbum.owner_id">owner_id</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.shared" href="#aioimmich.albums.models.ImmichAlbum.shared">shared</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.start_date" href="#aioimmich.albums.models.ImmichAlbum.start_date">start_date</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.to_dict" href="#aioimmich.albums.models.ImmichAlbum.to_dict">to_dict</a></code></li>
<li><code><a title="aioimmich.albums.models.ImmichAlbum.updated_at" href="#aioimmich.albums.models.ImmichAlbum.updated_at">updated_at</a></code></li>
</ul>
</li>
</ul>
</li>
</ul>
</nav>
</main>
<footer id="footer">
<p>Generated by <a href="https://pdoc3.github.io/pdoc" title="pdoc: Python API documentation generator"><cite>pdoc</cite> 0.11.6</a>.</p>
</footer>
</body>
</html>
