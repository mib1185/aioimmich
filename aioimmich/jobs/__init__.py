"""aioimmich jobs api."""

from ..api import ImmichSubApi
from .models import (
    ImmichJobsStatus,
    JobCommand,
    JobStatus,
)


class ImmichJobs(ImmichSubApi):
    """Immich jobs api."""

    async def async_get_all_jobs_status(self) -> ImmichJobsStatus:
        """Get all jobs status.

        Returns:
            all jobs status as `ImmichJobsStatus`
        """
        result = await self.api.async_do_request("jobs/status")
        assert isinstance(result, dict)
        return ImmichJobsStatus.from_dict(result)

    async def async_send_job_command(
        self, job_id: str, command: str, force: bool = False
    ) -> JobStatus:
        """Send job command.

        Args:
            job_id: The job type ID (e.g., "thumbnailGeneration", "metadataExtraction")
            command: The command to send ("start", "pause", "resume", "empty", "clear-failed")
            force: Whether to force the command

        Returns:
            job status as `JobStatus`
        """
        job_command = JobCommand(command=command, force=force)
        result = await self.api.async_do_request(
            f"jobs/{job_id}", data=job_command.to_dict(), method="PUT"
        )
        assert isinstance(result, dict)
        return JobStatus.from_dict(result)
