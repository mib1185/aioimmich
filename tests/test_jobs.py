"""Tests for aioimmich jobs."""

from __future__ import annotations

from syrupy.assertion import SnapshotAssertion

from aioimmich.jobs.models import ImmichAllJobStatus, ImmichJobStatus


async def test_get_all_jobs_status(mock_immich_with_data, snapshot: SnapshotAssertion):
    """Test async_get_all_jobs_status."""
    api = await mock_immich_with_data()
    jobs_status = await api.jobs.async_get_all_jobs_status()

    assert isinstance(jobs_status, ImmichAllJobStatus)
    assert jobs_status == snapshot


async def test_send_job_command(mock_immich_with_data, snapshot: SnapshotAssertion):
    """Test async_send_job_command."""
    api = await mock_immich_with_data()
    job_status = await api.jobs.async_send_job_command(
        "thumbnailGeneration", "pause", force=True
    )

    assert isinstance(job_status, ImmichJobStatus)
    assert job_status == snapshot
