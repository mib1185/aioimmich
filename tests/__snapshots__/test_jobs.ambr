# serializer version: 1
# name: test_get_all_jobs_status
  ImmichJobsStatus(background_task=JobStatus(job_counts=JobCounts(active=0, completed=1532, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), backup_database=JobStatus(job_counts=JobCounts(active=0, completed=45, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), duplicate_detection=JobStatus(job_counts=JobCounts(active=0, completed=0, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), face_detection=JobStatus(job_counts=JobCounts(active=0, completed=27038, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), facial_recognition=JobStatus(job_counts=JobCounts(active=0, completed=0, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), library=JobStatus(job_counts=JobCounts(active=0, completed=0, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), metadata_extraction=JobStatus(job_counts=JobCounts(active=0, completed=28874, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), migration=JobStatus(job_counts=JobCounts(active=0, completed=0, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), notifications=JobStatus(job_counts=JobCounts(active=0, completed=0, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), search=JobStatus(job_counts=JobCounts(active=0, completed=28874, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), sidecar=JobStatus(job_counts=JobCounts(active=0, completed=0, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), smart_search=JobStatus(job_counts=JobCounts(active=0, completed=0, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), storage_template_migration=JobStatus(job_counts=JobCounts(active=0, completed=0, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), thumbnail_generation=JobStatus(job_counts=JobCounts(active=0, completed=30924, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)), video_conversion=JobStatus(job_counts=JobCounts(active=0, completed=1836, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False)))
# ---
# name: test_send_job_command
  JobStatus(job_counts=JobCounts(active=0, completed=30924, delayed=0, failed=0, paused=0, waiting=0), queue_status=QueueStatus(is_active=True, is_paused=False))
# ---